lib/Db/PimcoreExtensionsTrait.php

```diff
@@ -611,4 +611,25 @@ public function setAutoQuoteIdentifiers($autoQuoteIdentifiers)
     {
         $this->autoQuoteIdentifiers = $autoQuoteIdentifiers;
     }
+
+    /**
+     * @param string $table
+     * @param string $idColumn
+     * @param string $where
+     */
+    public function selectAndDeleteWhere($table, $idColumn = 'id', $where = '')
+    {
+        $sql = 'SELECT ' . $idColumn . '  FROM ' . $table;
+
+        if ($where) {
+            $sql .= ' WHERE ' . $where;
+        }
+
+        $idsForDeletion = $this->fetchCol($sql);
+
+        if (!empty($idsForDeletion)) {
```

Comment:
Maybe we should think about processing them in chunks of maybe 50 IDs, to not exceed the max. query length `max_allowed_packet`.

Reply:
default value is 67108864 so I think 50 is a bit pesimistic

Reply:
😁 alright, maybe you're right. But splitting it into chunks makes sense, right?